### PR TITLE
Closes #2274: Refactor bundle tile logic

### LINKED ISSUE #2274: Identify storage format for new category channels
Blocks implementation of new channels to homescreen like #2196.

### Vision statement / What / Requirements
The new homescreen channels need to bundle tiles so we'll need to figure out how we want to store them.

**Keep it simple** and consider how it interacts with our existing bundled tile format.

### PR DESCRIPTION
**Pull request by psymoon**
### Pull Request checklist

- [ ] This PR includes thorough **tests** or an explanation of why it does not
- [ ] This PR includes a **CHANGELOG entry** or does not need one
- [ ] I have considered adding **QA labels** on the associated issue (not this PR; `qa-ready` or `qa-denied`)

### CODE REVIEW

File: app/src/main/java/org/mozilla/tv/firefox/channels/BundleTilesStore.kt [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+enum class BundleType {
+    PINNED_TILES
+}
+
+/**
+ * [BundleTilesStore] is responsible for fetching bundled tiles data from /assets/ with
+ * [BundleType]
+ */
+class BundleTilesStore(private val applicationContext: Application) {
+    fun getBundledTiles(type: BundleType): JSONArray {
```
Comment (Baron-Severin) [inline app/src/main/java/org/mozilla/tv/firefox/channels/BundleTilesStore.kt]: While we're moving things around anyway, I feel like we should be doing type conversions inside this class.  We lose a lot of type safety and clarity by exposing a `JSONArray`, I'd rather do that conversion on the edge between the app and storage (that is, here).

TL;DR: I think this should be a private function, then we can expose a `getPinnedTiles` method that calls this, converts, and returns a `List<ChannelTile>` (or `List<PinnedTile>` if that's unworkable).

File: app/src/main/java/org/mozilla/tv/firefox/channels/BundleTilesStore.kt [hunk 2]
```diff
@@ -19,12 +26,50 @@ enum class BundleType {
...
+            }
+        }
+
+        return tilesJSONArray
+    }
+
+    /**
+     * Used to handle removing bundle tiles by adding to its [BundleType] blacklist in SharedPreferences
+     */
+    fun addBundleTileToBlackList(type: BundleType, id: String) {
```
Comment (Baron-Severin) [inline app/src/main/java/org/mozilla/tv/firefox/channels/BundleTilesStore.kt]: I like this solution.

File: app/src/main/java/org/mozilla/tv/firefox/channels/pinnedtile/PinnedTileRepo.kt [hunk 3]
```diff
@@ -30,7 +29,7 @@ private const val PREF_HOME_TILES = "homeTiles"
  * @constructor loads the initial [_pinnedTiles] (a combination of custom and bundled tiles)
  */
 class PinnedTileRepo(
-    private val applicationContext: Application,
+    private val applicationContext: Context,
```
Comment (Baron-Severin) [inline app/src/main/java/org/mozilla/tv/firefox/channels/pinnedtile/PinnedTileRepo.kt]: Nit: If we're moving this to `Context`, we should rename it to match.

File: app/src/main/java/org/mozilla/tv/firefox/channels/BundleTilesStore.kt [hunk 4]
```diff
@@ -25,7 +24,7 @@ enum class BundleType {
  * [BundleTilesStore] is responsible for fetching bundled tiles data from /assets/ with
  * [BundleType]
  */
-class BundleTilesStore(private val applicationContext: Application) {
+class BundleTilesStore(private val applicationContext: Context) {
```
Comment (Baron-Severin) [inline app/src/main/java/org/mozilla/tv/firefox/channels/BundleTilesStore.kt]: Same as above

> Nit: If we're moving this to Context, we should rename it to match.

File: app/src/main/java/org/mozilla/tv/firefox/channels/BundleTilesStore.kt (hunk 1 above)
Comment (psymoon) [reply app/src/main/java/org/mozilla/tv/firefox/channels/BundleTilesStore.kt]: Yes I agree with this, I'll give some think on it
